vsdownload.py

```diff
@@ -203,14 +205,25 @@ def getManifest(args):
 
     return manifest
 
-def prioritizePackage(a, b):
-    if "chip" in a and "chip" in b:
-        ax64 = a["chip"].lower() == "x64"
-        bx64 = b["chip"].lower() == "x64"
-        if ax64 and not bx64:
+def prioritizePackage(arch, a, b):
+    def archOrd(k, x):
+        if arch is None:
+            return 0
+        c = x.get(k)
+        if c is None:
+            return 0
+        c = c.lower()
+        if c == "neutral":
             return -1
-        elif bx64 and not ax64:
-            return 1
+        if c == arch:
+            return -2
+        return 1
+    
+    for k in ["chip", "machineArch", "productArch"]:
```

Comment:
Can you briefly explain what roles these fields, `chip`, `machineArch` and `productArch` have wrt the tools? Are they all about the host architecture for where the tools are going to run, or is some aspect of these also about the intended target architecture they're producing code for?

Reply:
In fact, I can't. I think these fields are all about the architecture of binaries in the packages, and different packages use different fields. But we can't know whether **the package is for host machine or target machine**. See https://github.com/mstorsjo/msvc-wine/issues/135#issuecomment-2371844447.

Reply:
OK. We should treat them as host architecture. User can use `--host-arch` to download any architecture they want, no matter for host machine or target machine. So maybe `--machine-arch` is a better name? @mstorsjo any preference?

Reply:
I think `--host-arch` is a name that is easier to understand for most users - and in the vast majority of cases, users don't need to set the option at all. (And for the cases where you do want to set it, I guess it can be understood as "I want to download tools for host X", not necessarily "I'm running on host X right now".) So I think the current option name is fine - unless we at a later point find a reason why we should change it.

Reply:
Fair enough.